### DISCUSSION TITLE
Names in Version 3

### DISCUSSION THREAD

**Discussion by pvanthony** [romcal/romcal#317] [Q&A] [answered]
In version 2, the names were like "Saints Simon and Jude, Apostles". 
In version 3, the names are like this "simon_and_jude_apostles".

Looked into the array of the year and I could not find the name like "Saints Simon and Jude, Apostles".
Did I miss something or called a wrong array element?

On another note, version 3 is really fast compared to version 2. You guys must have done some magic code. Really fast. 
Now I can show three years of data and the site is almost instant. Cool!!!

**Comment by tukusejssirs** [comment]
@pvanthony, see [this comment](https://github.com/romcal/romcal/issues/299#issuecomment-933883512) and the entire issue.

Basically, I had the same question a while ago. The runtime speed-up is partially because the localisation is lazily computed. As @emagnier said:

> So they are still accessible, but their values are computed only when you access these properties. In fact, everything related to l10n / i18n are costly in terms of performance, this is the reason why they are not computed by default.

Example (just the essential commands; it does not work without other commands):

```js
const romcal = new Romcal({localizedCalendar: Slovakia_Sk})
const r = await romcal.generateCalendar(2021)
console.log(r['2021-01-01'][0].colorNames) // output [ "biela" ] in Slovak, it would be [ "white" ] in English
```

**Reply by pvanthony** [reply]
I tried with this code but the names are coming like something_something_something instead of Something Something Something.
```
            for(day in result){
                if(result[day][0].rank == "MEMORIAL"){
                    console.log("Memorial: ", result[day][0].date, result[day][0].name, result[day][0].colorNames)
                }
            }
```

Is there another word for name? Right now I am using result[day][0].name.

**Reply by emagnier** [reply]
Can you also provide the lines of code before, how you use romcal and fetch romcal data? 
This will be useful to help you better.

**Reply by pvanthony** [reply]
Thank you for helping.
```js
// create object romcal
const romcal = new Romcal({
    calendar: 'general',
    scope: 'gregorian',
    locale: currentLocale,
    epiphanyOnSunday: true,
    corpusChristiOnSunday: true,
    ascensionOnSunday: false,
    outputOptionalMemorials: false,
    strictMode: true
});

romcal.generateCalendar(today.getFullYear() + i).then(function(result) {

            console.log(result)

            for(day in result){
                if(result[day][0].rank == "MEMORIAL"){
                    console.log("Memorial: ", result[day][0].date, result[day][0].name)
                }
            }
}

```
Here is the link to the test site.
https://romcal.sound.sg/auto
In the console, the code result can be seen.

The javascript file is https://romcal.sound.sg/auto/packages/julian_custom.js

**Reply by tukusejssirs** [reply]
@pvanthony, here’s a working example. Read the code comments please. :wink:

Note that I used the following two scripts (change the paths as required):

```html
<script src='iife/romcal.js'></script>
<script src='bundles/sri-lanka/iife/en.js'></script>
```

```js
const romcal = new Romcal({
  // Note: `calendar` and `locale` are replaced by `localizedCalendar` in romcal v3.
  // calendar: 'general',
  // locale: currentLocale,
  localizedCalendar: SriLanka_En,
  scope: 'gregorian',
  epiphanyOnSunday: true,
  corpusChristiOnSunday: true,
  ascensionOnSunday: false,
  outputOptionalMemorials: false,
  strictMode: true
});

// Note: For testing purposes, I replaced `today.getFullYear() + i` with `new Date().getFullYear()` from year definition
romcal.generateCalendar(new Date().getFullYear()).then(result => {
  console.log(result)

  for (day in result) {
    if (result[day][0].rank === 'MEMORIAL') {
      // Note: Comma (`,`) in `console.log()` inserts a space, therefore there are two spaces after `Memorial:`.
      console.log('Memorial: ', result[day][0].date, result[day][0].name)
    }
  }
})
```

**Reply by pvanthony** [reply]
Thank you very much for the example code.
What is the config for general calendar? Instead of localizedCalendar: SriLanka_En.
Is it localizedCalendar: General_En?

**Reply by tukusejssirs** [reply]
In order to get the available calendar names, you can use `Romcal.CALENDAR_VAR_NAMES`. That is the first part of `localizedCalendar`.

Second part is a locale name. List of available locale names: `Romcal.LOCALE_VAR_NAMES`.

In order to answer your question, you can use `GeneralRoman_En` for the GRC in English (US) locale.

**Reply by pvanthony** [reply]
Thank you very much for advice and answer. I will try them out.

**Comment by pvanthony** [comment]
Yahooooooo! The names are working now!
Thank you. Thank you Thank you for helping and being patient with me.
Both of you have been very helpful.

**Reply by tukusejssirs** [reply]
Good.

I don’t want to be picky, but you marked your last comment as the _answer_, however, I don’t think that that particular comment answers you original question. I suggest to change that. :wink:

**Reply by pvanthony** [reply]
Ops! I have removed the answered tag now.
please let me know how to mark the actual answer as answered.

**Reply by tukusejssirs** [reply]
I’ve just noticed that GitHub allows only the top-level comments marked as answers (see github/feedback#3892). Therefore we (you) have currently two options:
- mark one of the top-level comments as an anwer;
- create a new top-level comment with the answer and mark it as an anwser (you might provide a link to the comment with the original answer which helped you).

**Comment by pvanthony** [comment, marked as answer]
Here is the solution that was used.
https://github.com/romcal/romcal/discussions/317#discussioncomment-1551447
Thanks again for helping. Really do appreciate it very much. Especially for a non programmer like me.
